src/apps/vpn/connectionhealth.cpp

```diff
@@ -216,13 +193,17 @@ void ConnectionHealth::dnsPingReceived(quint16 sequence) {
   if (sequence != m_dnsPingSequence) {
     return;
   }
+
   quint64 latency = QDateTime::currentMSecsSinceEpoch() - m_dnsPingTimestamp;
   logger.debug() << "Received DNS ping:" << latency << "msec";
+  updateDnsPingLatency(latency);
+}
 
+void ConnectionHealth::updateDnsPingLatency(quint64 latency) {
   if (m_dnsPingInitialized) {
     m_dnsPingLatency *= (PING_BASELINE_EWMA_DIVISOR - 1);
     m_dnsPingLatency += latency;
-    m_dnsPingLatency /= PING_BASELINE_EWMA_DIVISOR;
+    m_dnsPingLatency = round(static_cast<float>(m_dnsPingLatency) / PING_BASELINE_EWMA_DIVISOR);
```

Comment:
This is a bit pendatic on my part. We were just truncating before, happy to reset this.